apps/minifront/src/components/ibc/ibc-out/chain-selector.tsx

```diff
@@ -7,23 +7,27 @@ import {
 } from '@repo/ui/components/ui/select';
 import { cn } from '@repo/ui/lib/utils';
 import { useState } from 'react';
-import { useStore } from '../../../state';
-import { ibcOutSelector } from '../../../state/ibc-out';
-import { useLoaderData } from 'react-router-dom';
-import { IbcLoaderResponse } from '../ibc-loader';
+import { AllSlices } from '../../../state';
 import { Chain } from '@penumbra-labs/registry';
+import { useStoreShallow } from '../../../utils/use-store-shallow';
+import { useChains } from '../../../state/ibc-out';
+
+const chainSelectorSelector = (state: AllSlices) => ({
+  chain: state.ibcOut.chain,
+  setChain: state.ibcOut.setChain,
+});
 
 export const ChainSelector = () => {
-  const { chain, setChain } = useStore(ibcOutSelector);
-  const { chains: ibcConnections } = useLoaderData() as IbcLoaderResponse;
+  const { chain, setChain } = useStoreShallow(chainSelectorSelector);
+  const chains = useChains();
```

Comment:
This is generated by the new ZQuery `chains` object in the `ibcOut` slice

Reply:
Also, I found it confusing that we're calling them `ibcConnections` in some places and `chains` in others, so I normalized on `chains`